app/grandchallenge/cases/image_builders/tiff.py

```diff
@@ -248,37 +244,48 @@ def image_builder_tiff(path: Path, session_id=None) -> ImageBuilderResult:
     )
 
 
-def create_tiff_image_entry(*, tiff_file: GrandChallengeTiffFile, pk) -> Image:
+def create_tiff_image_entry(
+    tiff_file: GrandChallengeTiffFile, pk: UUID
+) -> Image:
     # Builds a new Image model item
     return Image(
         pk=pk,
         name=tiff_file.path.name,
-        width=tiff_file.tags.image_width,
-        height=tiff_file.tags.image_height,
+        width=tiff_file.image_width,
+        height=tiff_file.image_height,
         depth=1,
-        resolution_levels=tiff_file.tags.resolution_levels,
-        color_space=tiff_file.tags.color_space,
+        resolution_levels=tiff_file.resolution_levels,
+        color_space=tiff_file.color_space,
         eye_choice=Image.EYE_UNKNOWN,
         stereoscopic_choice=Image.STEREOSCOPIC_UNKNOWN,
         field_of_view=Image.FOV_UNKNOWN,
-        voxel_width_mm=tiff_file.tags.voxel_width_mm,
-        voxel_height_mm=tiff_file.tags.voxel_height_mm,
-        voxel_depth_mm=tiff_file.tags.voxel_depth_mm,
+        voxel_width_mm=tiff_file.voxel_width_mm,
+        voxel_height_mm=tiff_file.voxel_height_mm,
+        voxel_depth_mm=tiff_file.voxel_depth_mm,
     )
 
 
-def create_dzi_images(*, tiff_file: GrandChallengeTiffFile, pk) -> str:
+def create_dzi_images(
+    gc_file: GrandChallengeTiffFile, pk: UUID
+) -> (str, GrandChallengeTiffFile):
```

Comment:
Please go back to the keyword only function, I have a strong preference for this as it makes the code much more maintainable and grok-able, happy to discuss this decision.

Reply:
Ok. For all functions or just the public ones?

Reply:
For both changed in this PR at least, may be more work to get the others to be keyword only and doesn't have priority right now

Reply:
Not sure why the tests fail now, do I need to do something more than adding the libopenslide-dev to the dockerfile?